independent-projects/qute/core/src/main/java/io/quarkus/qute/ValueResolvers.java

```diff
@@ -445,7 +452,14 @@ private static Object takeLastArray(int n, Object sourceArray) {
                             });
                 }
             default:
-                return Results.notFound(context);
+                // Try to use the name as an index
+                int index;
+                try {
+                    index = Integer.parseInt(name);
+                } catch (NumberFormatException e) {
```

Comment:
Can't `name` be null and throw an NPE here?

Reply:
Nope, `EvalContext.getName()` should never return `null`.